File: pwbreach.py
import hashlib
import requests

def get_pwned_passwords(prefix):
    url = f"https://api.pwnedpasswords.com/range/{prefix}"
    response = requests.get(url)
    
    if response.status_code != 200:
        raise RuntimeError(f"Error fetching data from HIBP API: {response.status_code}")
    
    return response.text

def check_password(password):
    sha1_hash = hashlib.sha1(password.encode()).hexdigest().upper()
    prefix, suffix = sha1_hash[:5], sha1_hash[5:]

    breached_hashes = get_pwned_passwords(prefix)

    print(prefix)
    print(breached_hashes)

    for line in breached_hashes.splitlines():
        hash_suffix, count = line.split(':')
        if hash_suffix == suffix:
            return int(count)
    
    return 0

if __name__ == "__main__":
    password = input("Entar Password : ")
    count = check_password(password)

    if count:
        print(f"⚠️ Password has been breached {count} times! Please change the password!!!")
    else:
        print("✅ Password is safe, not found in leaked database.")

File: pwbreach.php
<?php
function get_pwned_passwords($prefix) {
    $url = "https://api.pwnedpasswords.com/range/" . strtoupper($prefix);
    $response = file_get_contents($url);

    if ($response === FALSE) {
        die("Error fetching data from HIBP API.");
    }

    return $response;
}

function check_password($prefix) {
    $breached_hashes = get_pwned_passwords($prefix);

    // Format response menjadi array
    $hash_list = explode("\n", $breached_hashes);

    // Buat output JSON
    $result = [];
    foreach ($hash_list as $line) {
        list($hash_suffix, $count) = explode(":", trim($line));
        $result[] = [
            "full_hash" => strtoupper($prefix . $hash_suffix),
            "breach_count" => intval($count)
        ];
    }

    return json_encode($result, JSON_PRETTY_PRINT);
}

// Ambil input dari parameter GET
if (isset($_GET['hash'])) {
    $hash_prefix = $_GET['hash'];

    // Validasi apakah hash terdiri dari 5 karakter heksadesimal
    if (!preg_match('/^[a-fA-F0-9]{5}$/', $hash_prefix)) {
        die(json_encode(["error" => "Invalid hash prefix. Must be exactly 5 hex characters."]));
    }

    // Cek hash di HIBP API
    header('Content-Type: application/json');
    echo check_password($hash_prefix);
} else {
    die(json_encode(["error" => "No hash prefix provided. Use ?hash=XXXXX"]));
}
?>
